nbformat/validator.py

```diff
@@ -241,21 +242,35 @@ def validate(nbjson, ref=None, version=None, version_minor=None, relax_add_props
     Raises ValidationError if not valid.
     """
     if version is None:
-        from .reader import get_version
-        (version, version_minor) = get_version(nbjson)
+        version, version_minor = get_version(nbjson)
+
+    try:
+        err = next(iter_validate(nbjson, ref=ref, version=version, version_minor=version_minor, relax_add_props=relax_add_props))
+        raise err
+    except StopIteration:
+        return None
+
+
+def iter_validate(nbjson, ref=None, version=None, version_minor=None, relax_add_props=False):
+    """Checks whether the given notebook JSON conforms to the current
+    notebook format schema.
+
+    Returns a generator of all ValidationErrors if not valid.
+    """
+    if version is None:
+        version, version_minor = get_version(nbjson)
 
     validator = get_validator(version, version_minor, relax_add_props=relax_add_props)
 
     if validator is None:
         # no validator
         warnings.warn("No schema for validating v%s notebooks" % version, UserWarning)
-        return
+        raise StopIteration
```

Comment:
I think this can stay a `return` to terminate the iteration, without raising StopIteration which is deprecated.

Reply:
I didn't know that StopIteration was deprecated.  Thanks for pointing that out.